worlds/cvcotm/aesthetics.py

```diff
@@ -0,0 +1,779 @@
+                rare_drops[i] = select_drop(world, HIGH_ITEMS, placed_high_items)
+
+            # Set base drop chances at 6-10% for common and 3-6% for rare.
+            regular_drop_chances[i] = 600 + world.random.randint(0, 400)
+            rare_drop_chances[i] = 300 + world.random.randint(0, 300)
+
+        # Regular enemies outside Tiered logic.
+        else:
+            # Select a random regular and rare drop for every enemy from their respective lists.
+            regular_drops[i] = select_drop(world, COMMON_ITEMS, placed_common_items)
+            rare_drops[i] = select_drop(world, RARE_ITEMS, placed_rare_items)
+
+            # Set base drop chances at 6-10% for common and 3-6% for rare.
+            regular_drop_chances[i] = 600 + world.random.randint(0, 400)
+            rare_drop_chances[i] = 300 + world.random.randint(0, 300)
+
+    # Return the randomized drop data as bytes with their respective offsets.
+    enemy_address = ENEMY_TABLE_START
+    drop_data = {}
+    for i in range(NUMBER_ENEMIES):
+        drop_data[enemy_address] = bytes([regular_drops[i], 0, regular_drop_chances[i] & 0xFF,
+                                          regular_drop_chances[i] >> 8, rare_drops[i], 0, rare_drop_chances[i] & 0xFF,
+                                          rare_drop_chances[i] >> 8])
+        enemy_address += 20
+
+    return drop_data
+
+
+def select_drop(world: "CVCotMWorld", drop_list: List[int], drops_placed: List[int], exclusive_drop: bool = False,
+                start_index: int = 0) -> int:
+    """Chooses a drop from a given list of drops based on another given list of how many drops from that list were
+    selected before. In order to ensure an even number of drops are distributed, drops that were selected the least are
+    the ones that will be picked from.
+
+    If Item Drop Randomization is set to Hard, calling this with True as the exclusive_drop param will force the number
+    of the chosen item really high to ensure it will never be picked again."""
+
+    number_valid_drops = 0
+    eligible_items = [0] * NUMBER_ITEMS
+    lowest_number = drops_placed[start_index]
+
+    # Only make eligible drops which we have placed the least.
+    i = start_index
+    while i < len(drop_list):
+        # A drop with the priority we are expecting is available to add as a candidate.
+        if drops_placed[i] == lowest_number:
+            eligible_items[number_valid_drops] = i
+            number_valid_drops += 1
+            i += 1
+
+        # If this condition is met, there is at least one item that hasn't been placed as many times as the others.
+        # We have to lower the lowest number and start from the beginning of the loop to capture all the valid indices.
+        elif drops_placed[i] < lowest_number:
+            lowest_number = drops_placed[i]
+            number_valid_drops = 0
+            i = start_index
+
+        else:
+            i += 1
```

Comment:
What you're looking at right there is my attempt at turning this C code from upstream into Python code, lmao:
![Screenshot 2024-11-30 004818](https://github.com/user-attachments/assets/e7a894b5-0c74-4e9c-afc5-4bd48692f8fe)
I'll try your suggestion.

Reply:
Glad I decided to actually check what the final placement count list variables were by the time the randomization was done they were looking like this:
![Screenshot 2024-11-30 064813](https://github.com/user-attachments/assets/b4e0736b-27c8-4a52-aa69-0bde8c5885ec)
The expected output is something like this (for Normal drop randomization):
![Screenshot 2024-11-30 065607](https://github.com/user-attachments/assets/f09a0fea-bb14-4b12-a62d-75add216f70b)


Should be fixed now.

Reply:
I made the mistake of just looking at the counts and not testing in-game because I was in such a hurry earlier to get out the door then, and it turned out that the code was not returning the correct item drop IDs, WHOOPS!

Checked the placement count lists *and* checked in-game, and now I am very confident it's working.